expression/builtin_info.go

```diff
@@ -771,6 +776,122 @@ func (k TiDBDecodeKeyFunctionKeyType) String() string {
 
+type tidbDecodeSQLDigestsFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *tidbDecodeSQLDigestsFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+
+	var argTps []types.EvalType
+	if len(args) > 1 {
+		argTps = []types.EvalType{types.ETString, types.ETInt}
+	} else {
+		argTps = []types.EvalType{types.ETString}
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, argTps...)
+	if err != nil {
+		return nil, err
+	}
+	sig := &builtinTiDBDecodeSQLDigestsSig{bf}
+	return sig, nil
+}
+
+type builtinTiDBDecodeSQLDigestsSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinTiDBDecodeSQLDigestsSig) Clone() builtinFunc {
+	newSig := &builtinTiDBDecodeSQLDigestsSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+func (b *builtinTiDBDecodeSQLDigestsSig) evalString(row chunk.Row) (string, bool, error) {
+	args := b.getArgs()
+	digestsStr, isNull, err := args[0].EvalString(b.ctx, row)
+	if err != nil {
+		return "", true, err
+	}
+	if isNull {
+		return "", true, nil
+	}
+
+	stmtTruncateLength := int64(0)
+	if len(args) > 1 {
+		stmtTruncateLength, isNull, err = args[1].EvalInt(b.ctx, row)
+		if err != nil {
+			return "", true, err
+		}
+		if isNull {
+			stmtTruncateLength = 0
+		}
+	}
+
+	var digests []interface{}
+	err = json.Unmarshal([]byte(digestsStr), &digests)
+	if err != nil {
+		const errMsgMaxLength = 32
+		if len(digestsStr) > errMsgMaxLength {
+			digestsStr = digestsStr[:errMsgMaxLength] + "..."
+		}
+		b.ctx.GetSessionVars().StmtCtx.AppendWarning(errIncorrectArgs.GenWithStack("The argument can't be unmarshalled as JSON array: '%s'", digestsStr))
+		return "", true, nil
+	}
+
+	// Query the SQL Statements by digests.
+	retriever := NewSQLDigestTextRetriever()
+	for _, item := range digests {
+		if item != nil {
+			digest, ok := item.(string)
+			if ok {
+				retriever.SQLDigestsMap[digest] = ""
+			}
+		}
+	}
+
+	// Querying may take some time and it takes a context.Context as argument, which is not available here.
+	// We simply create a context with 1s timeout here.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	err = retriever.RetrieveGlobal(ctx, b.ctx)
+	if err != nil {
+		if errors.Cause(err) == context.DeadlineExceeded || errors.Cause(err) == context.Canceled {
+			return "", true, errUnknown.GenWithStack("Retrieving cancelled internally with error: %v", err)
+		}
+
+		b.ctx.GetSessionVars().StmtCtx.AppendWarning(errUnknown.GenWithStack("Retrieving statements information failed with error: %v", err))
+		return "", true, nil
+	}
+
+	// Collect the result.
+	result := make([]interface{}, len(digests))
+	for i, item := range digests {
+		if item == nil {
+			continue
+		}
+		if digest, ok := item.(string); ok {
+			if stmt, ok := retriever.SQLDigestsMap[digest]; ok && len(stmt) > 0 {
```

Comment:
What will be displayed if some digests are not found?

Reply:
null

Reply:
If some items in the json array are not strings, they will also be mapped to null. You can see these cases in the unit test.